### PR TITLE
fix(web): surface runtime state in Tools modal and MCP Servers settings

### PR DESCRIPTION
**Pull request by lsm**
Tools modal now shows a per-skill runtime indicator derived from the session MCP list (active / server-off with scope source / server-missing / skill-disabled / unknown-while-loading), so users can tell when a "checked" skill is not actually being injected.

MCP Servers settings links each `app_mcp_servers` row to its skill wrapper; orphan rows render an amber warning that toggling them has no runtime effect.

Runtime-state and linkage logic live in `ToolsModal.utils.ts` so tests exercise the same code the UI uses.

Addresses Task #114.

### CODE REVIEW

**Review by lsm — COMMENTED** [review]
## Review: fix(web): surface runtime state in Tools modal and MCP Servers settings

The overall approach is solid — extracting the state-computation logic into pure functions in `ToolsModal.utils.ts`, testing those functions directly, and wiring them into both the modal and the settings panel. The `sessionMcpLoaded` signal in `ToolsModal.tsx` is the right pattern to avoid the "no entries" vs "not yet loaded" ambiguity. Comments and docstrings are accurate.

Two issues need fixing before merge.

---

### 🔴 Issue 1 — `skillsLoaded` proxy is semantically wrong (`AppMcpServersSettings.tsx` line 360)

```tsx
const skillsLoaded = skills.length > 0;
```

This reads "at least one skill exists" but is used as "skills have been fetched." These diverge in any installation where the space has MCP servers but **zero skills of any type** — the orphan warnings (`⚠️ No skill wrapper…`) would never render, even after the LiveQuery snapshot has arrived. The proxy happens to work in practice because built-in skills (playwright, etc.) are always present, but it's a correctness gap that will silently bite a minimal/clean install.

`ToolsModal.tsx` handles the analogous problem correctly with an explicit `sessionMcpLoaded` signal. `AppMcpServersSettings` should follow the same pattern — either expose a `loaded` signal from `SkillsStore` (simplest) or track it locally with a `useRef/useState` that flips when `isLoading` transitions `true → false` after at least one render with `isLoading === true`.

The test named _"suppresses annotations while skills are still loading"_ passes today because it sets `skills.value = []`, which is the same value as the not-yet-loaded state — so it validates the proxy rather than the intended behaviour.

---

### 🟡 Issue 2 — `server-off` renders with a gray dot (same as `skill-disabled`), which is misleading (`ToolsModal.tsx` lines 685-696)

```tsx
const runtimeDotClass =
    runtime.status === 'active'
        ? 'bg-emerald-400'
        : runtime.status === 'server-missing'
            ? 'bg-red-400'
            : 'bg-gray-500';   // ← covers both server-off AND skill-disabled
```

`skill-disabled` (user explicitly unchecked the skill) and `server-off` (skill is checked but its backing server is disabled at a scope the user didn't touch) look identical: gray dot + gray text. The acceptance criterion says the UI should "clearly communicate when a skill is enabled in the registry vs confirmed active in session" — a gray dot for `server-off` doesn't meet that bar.

`server-off` should be amber (`bg-amber-400` / `text-amber-500/70`) to match the warning affordance used elsewhere in the settings panel (the orphan-warning text in `AppMcpServersSettings` uses `text-amber-400/90`). `skill-disabled` can stay gray since the user made that choice deliberately.

---

### Minor / non-blocking

- The `import { computeMcpSkillRuntimeState, computeSkillGroupState } from './ToolsModal.utils.ts'` at line 65 of `ToolsModal.tsx` sits after the `RUNTIME_MCP_LABELS` constant block. That's a pre-existing misplacement but the PR touched this import — worth hoisting it to join the other imports.
- The orphan-warning div in `AppMcpServersSettings` (line 407-410) does not use `truncate` while the skill-link div above it does. On very narrow viewports the long warning string ("⚠️ No skill wrapper — this server is not injected into any session. Toggling it here has no runtime effect.") will wrap multiple lines inside an already compact row. Either add `truncate` and rely on the `title` attribute, or cap with a `max-w-*` + `truncate`.

File: packages/web/src/components/settings/AppMcpServersSettings.tsx [hunk 1]
```diff
@@ -324,79 +351,106 @@ export function AppMcpServersSettings() {
...
-												server.sourceType === 'stdio' && 'bg-green-500/20 text-green-400',
-												server.sourceType === 'sse' && 'bg-blue-500/20 text-blue-400',
-												server.sourceType === 'http' && 'bg-purple-500/20 text-purple-400'
+						{servers.map((server) => {
+							// Show linkage info only after the skills subscription has loaded
+							// at least one row. Before that, we genuinely don't know whether a
+							// wrapper exists, so suppressing the annotation prevents a
+							// misleading "⚠️ Not exposed" flash during initial mount.
+							const linkedSkill = skillLinkage.get(server.id);
+							const skillsLoaded = skills.length > 0;
```
Comment (lsm) [inline packages/web/src/components/settings/AppMcpServersSettings.tsx]: skillsLoaded proxy is semantically incorrect. skills.length > 0 reads as 'at least one skill exists' but is used to mean 'the skills subscription has delivered its first snapshot.' These diverge when a space has MCP servers but zero skills — orphan warnings would never appear after load. ToolsModal.tsx solves the identical problem with an explicit sessionMcpLoaded signal. Please follow the same pattern: add a loaded boolean to SkillsStore (or track locally) that flips when the snapshot handler fires.

File: packages/web/src/components/ToolsModal.tsx [hunk 2]
```diff
@@ -660,6 +677,23 @@ export function ToolsModal({ isOpen, onClose, session }: ToolsModalProps) {
...
+													skill,
+													sessionMcpEntries.value,
+													sessionMcpLoaded.value
+												);
+												const runtimeDotClass =
+													runtime.status === 'active'
+														? 'bg-emerald-400'
+														: runtime.status === 'server-missing'
+															? 'bg-red-400'
+															: 'bg-gray-500';
```
Comment (lsm) [inline packages/web/src/components/ToolsModal.tsx]: server-off gets a gray dot (same as skill-disabled), which is misleading. When status is 'active' we show green; 'server-missing' shows red. But 'server-off' (the skill is checked but the backing server is disabled at a scope the user didn't explicitly set) looks identical to 'skill-disabled' (the user turned it off). These are meaningfully different situations. Please add an amber branch for 'server-off': bg-amber-400 / text-amber-500/70 — consistent with the amber warning colour already used in the orphan-warning text in AppMcpServersSettings.

File: packages/web/src/components/settings/AppMcpServersSettings.tsx (hunk 1 above)
Comment (lsm) [reply packages/web/src/components/settings/AppMcpServersSettings.tsx]: Fixed in f67360d0b. Added an explicit `loaded: signal<boolean>` to `SkillsStore` (flips to `true` in the `liveQuery.snapshot` handler on the first event; resets to `false` in `teardownCleanly`). `AppMcpServersSettings` now reads `skillsStore.loaded.value` instead of `skills.length > 0`, so a space with MCP servers and zero wrapper skills correctly surfaces the orphan warning once the subscription has delivered its (empty) snapshot.

Mirrors the `sessionMcpLoaded` pattern already used in `ToolsModal.tsx`, so the two views treat "subscription in flight" identically.

Split the original "suppresses annotations while loading" test into two:
- `suppresses ... while the skills subscription has not yet delivered a snapshot` (the `loaded === false` branch)
- `shows orphan warning once the subscription has loaded even if there are zero skills` — direct regression guard for the bug you flagged.

Also added three SkillsStore tests covering the `loaded` lifecycle (initial false, flips on empty snapshot, resets on unsubscribe).

File: packages/web/src/components/ToolsModal.tsx (hunk 2 above)
Comment (lsm) [reply packages/web/src/components/ToolsModal.tsx]: Fixed in f67360d0b. `server-off` now renders amber (`bg-amber-400` / `text-amber-500/70`), matching the orphan-warning amber in `AppMcpServersSettings`. `skill-disabled` stays gray (user's own choice), `active` stays emerald, `server-missing` stays red.

Also refactored the mapping out of the inline ternary chains into a pure util `getMcpSkillRuntimeClasses(status) → { dot, text }` in `ToolsModal.utils.ts`, so the colour semantics are documented in one place and unit-tested — with an explicit assertion that `server-off !== skill-disabled` as a regression guard.

**Review by lsm — COMMENTED** [review]
Both requested changes addressed correctly.

**Issue 1 (skillsLoaded proxy):** `SkillsStore` now exposes a `loaded` signal that flips in the snapshot handler and resets on teardown — exactly the right place. `AppMcpServersSettings` reads `skillsStore.loaded.value` instead of the length proxy. The three new store lifecycle tests are precise regression guards: one for the in-flight state, one for the zero-rows steady state (the exact bug), and one verifying the reset on unsubscribe. The two AppMcpServersSettings tests are correctly split and the test names now accurately describe what they test.

**Issue 2 (server-off color):** Extracting `getMcpSkillRuntimeClasses` into `ToolsModal.utils.ts` was the right call — the colour mapping is now documented (JSDoc colour semantics table) and unit-tested with explicit assertions that `server-off ≠ skill-disabled`. The amber choice is consistent with the orphan-warning amber in `AppMcpServersSettings`.

Implementation is complete. ✅
